src/rdpc/analysesProcessor.ts

```diff
@@ -277,3 +347,43 @@ const initializeRdpcInfo = (
     processingStatus: "REGISTERED",
   };
 };
+
+export const mergeDonorStateMaps = (
+  map: DonorRunStateMap,
+  mergeWith: DonorRunStateMap
+): DonorRunStateMap => {
+  const result = Object.entries(mergeWith).reduce<DonorRunStateMap>(
+    (acc, [donorId, rdpcInfo]) => {
+      acc[donorId] = {
+        publishedNormalAnalysis:
+          (acc[donorId]?.publishedNormalAnalysis || 0) +
+          rdpcInfo.publishedNormalAnalysis,
+        publishedTumourAnalysis:
+          (acc[donorId]?.publishedTumourAnalysis || 0) +
+          rdpcInfo.publishedTumourAnalysis,
+        alignmentsCompleted:
+          (acc[donorId]?.alignmentsCompleted || 0) +
+          rdpcInfo.alignmentsCompleted,
+        alignmentsRunning:
+          (acc[donorId]?.alignmentsRunning || 0) + rdpcInfo.alignmentsRunning,
+        alignmentsFailed:
+          (acc[donorId]?.alignmentsFailed || 0) + rdpcInfo.alignmentsFailed,
+        sangerVcsCompleted:
+          (acc[donorId]?.sangerVcsCompleted || 0) + rdpcInfo.sangerVcsCompleted,
+        sangerVcsRunning:
+          (acc[donorId]?.sangerVcsRunning || 0) + rdpcInfo.sangerVcsRunning,
+        sangerVcsFailed:
+          (acc[donorId]?.sangerVcsFailed || 0) + rdpcInfo.sangerVcsFailed,
+        totalFilesCount:
+          (acc[donorId]?.totalFilesCount || 0) + rdpcInfo.totalFilesCount,
+        filesToQcCount:
+          (acc[donorId]?.filesToQcCount || 0) + rdpcInfo.filesToQcCount,
+        releaseStatus: "NO_RELEASE",
+        processingStatus: "REGISTERED",
+      };
+      return acc;
+    },
+    { ...map }
```

Comment:
seeing this it's pretty tempting to remove the spread, but the spread is actually meaningful here to avoid mutating original `map` so... let's write this as `_.clone(map)` so the cloning intention is clearer